### ISSUE TITLE
CIMultiDict copy not working as expected

### ISSUE THREAD

**Issue by lsbardel** [aio-libs/multidict#107]
There is an issue with ``CIMultiDict`` copy method
```python
import multidict
multidict.__version__  # '3.1.1'
a = multidict.CIMultiDict()
a['foo'] = 6
b = a.copy()
b['foo'] = 7
b # <CIMultiDict('foo': 7)>
a # <CIMultiDict('foo': 7)>
```

And using the ``copy`` function
```python
import copy
c = copy.copy(a)
```
I get
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../venv/lib/python3.6/copy.py", line 106, in copy
    return _reconstruct(x, None, *rv)
  File ".../venv/lib/python3.6/copy.py", line 274, in _reconstruct
    y = func(*args)
  File "multidict/_multidict.pyx", line 531, in multidict._multidict.CIMultiDict.__init__ (multidict/_multidict.c:9519)
  File "multidict/_multidict.pyx", line 327, in multidict._multidict.MultiDict._extend (multidict/_multidict.c:6702)
TypeError: CIMultiDict takes either dict or list of (key, value) tuples
```

**Comment by lsbardel** [comment]
Copy initialisation has issues too (all related I assume):
```python
a = multidict.CIMultiDict(foo=6)
b = multidict.CIMultiDict(a)
b['foo'] = 7
b # <CIMultiDict('foo': 7)>
a # <CIMultiDict('foo': 7)>
```
but this works OK
```python
a = multidict.CIMultiDict(foo=6)
b = multidict.CIMultiDict(a.items())
b['foo'] = 7
b # <CIMultiDict('foo': 7)>
a # <CIMultiDict('foo': 6)>
```

**Comment by asvetlov** [comment]
Thanks for report!

**Comment by asvetlov** [comment]
Fixed by 3555d46 and 04e740d
